config/src/client_api.rs

```diff
@@ -0,0 +1,82 @@
+//! Functionality related to working with the configuration through client API.
+//!
+//! Intended usage:
+//!
+//! - Create [`ConfigurationDTO`] from [`crate::iroha::Configuration`] and serialize it for the client
+//! - Deserialize [`ConfigurationDTO`] from the client and use [`ConfigurationDTO::apply_update()`] to update the configuration
+// TODO: Currently logic here is not generalised and handles only `logger.max_log_level` parameter. In future, when
+//       other parts of configuration are refactored and there is a solid foundation e.g. as a general
+//       configuration-related crate, this part should be re-written in a clean way.
+//       Track configuration refactoring here: https://github.com/hyperledger/iroha/issues/2585
+
+use iroha_config_base::runtime_upgrades::{Reload, ReloadError};
+use iroha_data_model::Level;
+use serde::{Deserialize, Serialize};
+
+use super::{iroha::Configuration as BaseConfiguration, logger::Configuration as BaseLogger};
+
+/// Subset of [`super::iroha`] configuration.
+#[derive(Debug, Serialize, Deserialize, Clone, Copy)]
+pub struct ConfigurationDTO {
```

Comment:
Maybe call it `PartialConfiguration` or `DynamicConfiguration`?

Reply:
I thought that `DTO` suffix emphasizes the purpose to use this structure to interact with the client side.

`Partial` seems to be too general here (like `Proxy`s we have already), while `Dynamic` might semantically fit.